tests/integration/security/external_ca/main_test.go

```diff
@@ -91,41 +92,58 @@ func TestMain(m *testing.M) {
 }
 
 func setupConfig(ctx resource.Context, cfg *istio.Config) {
-	go csrctrl.RunCSRController("clusterissuers.istio.io/signer1", ctx.Clusters()[0].RESTConfig(), stopChan)
+	certsChan := make(chan *csrctrl.SignerRootCert, 2)
+	go csrctrl.RunCSRController("clusterissuers.istio.io/signer1,clusterissuers.istio.io/signer2", false,
+		ctx.Clusters()[0].RESTConfig(), stopChan, certsChan)
+	cert1 := <-certsChan
+	cert2 := <-certsChan
+
 	if cfg == nil {
 		return
 	}
-	cfg.ControlPlaneValues = `
-meshConfig:
-  defaultConfig:
-    proxyMetadata:
-      ISTIO_META_CERT_SIGNER: signer1
-components:
-  pilot:
-    k8s:
-      env:
-      - name: CERT_SIGNER_DOMAIN
-        value: clusterissuers.istio.io
-      - name: EXTERNAL_CA
-        value: ISTIOD_RA_KUBERNETES_API
-      overlays:
-        # Amend ClusterRole to add permission for istiod to approve certificate signing by custom signer
-        - kind: ClusterRole
-          name: istiod-clusterrole-istio-system
-          patches:
-            - path: rules[-1]
-              value: |
-                apiGroups:
-                - certificates.k8s.io
-                resourceNames:
-                - clusterissuers.istio.io/*
-                resources:
-                - signers
-                verbs:
-                - approve
+	cfgYaml := tmpl.MustEvaluate(`
 values:
   meshConfig:
+    defaultConfig:
+      proxyMetadata:
+        PROXY_CONFIG_XDS_AGENT: "true"
+        ISTIO_META_CERT_SIGNER: signer1
     trustDomainAliases: [some-other, trust-domain-foo]
-`
+    caCertificates:
+    - pem: |
+{{.rootcert1 | indent 8}}
+      certSigners:
+      - {{.signer1}}
+    - pem: |
+{{.rootcert2 | indent 8}}
+      certSigners:
+      - {{.signer2}}
+  components:
```

Comment:
Would be great to add a negative test case here - have the signer1's caCertificate mapped into signer2 in meshConfig and ensure that the cert Signing fails. This can be handled later.

Reply:
Let's add this  in later PR